lms/djangoapps/certificates/models.py

```diff
@@ -296,41 +298,51 @@ def __repr__(self):
             user=self.user
         )
 
-    def invalidate(self, source=None):
+    def invalidate(self, mode=None, source=None):
```

Comment:
Have we confirmed all callers are calling with source as a kwarg not a positional argument? (I don't see any in this PR) Otherwise we'll need to add it to the end of the arg list.

Reply:
Thanks for asking. Existing callers are using named arguments

* https://github.com/edx/edx-platform/blob/8a7ad0630412b1a5f841404574c537963abd59c5/lms/djangoapps/instructor/views/api.py#L3331
* https://github.com/edx/edx-platform/blob/3218ee4f1cfdd622956ca4cdda891ac532b06e05/lms/djangoapps/certificates/services.py#L38
* https://github.com/edx/edx-platform/blob/3218ee4f1cfdd622956ca4cdda891ac532b06e05/lms/djangoapps/certificates/generation_handler.py#L278